uservice_ccutter/server.py

```diff
@@ -1,61 +1,281 @@
+                allfiles = repo.untracked_files
+                idx = repo.index
+                idx.add(allfiles)
+                # Looks like userdict should include author email....
+                # We need consistent field names here...but we can
+                #  synthesize them in the plugin.
+                # git_name and git_email it is.
+                committer = git.Actor(userdict["github_name"],
+                                      userdict["github_email"])
+                idx.commit("Initial commit.", author=committer,
+                           committer=committer)
+                # Now we need to create the repository at Github....
+                #  userdict["github_repo"] must exist.
+                remote_url = create_github_repository(auth, userdict)
+                # Set up remote config to auth correctly
+                chlp = get_git_credential_helper(auth["username"],
+                                                 auth["password"])
+                origin = repo.create_remote("origin", url=remote_url)
+                cwr = repo.config_writer()
+                if not cwr.has_section("credential"):
+                    cwr.add_section("credential")
+                cwr.set("credential", "helper", chlp)
+                cwr.release()
+                # https://gitpython.readthedocs.io/en/stable/tutorial.html
+                #  suggests that you need to wait/sync or something?
+                time.sleep(1)
+                try:
+                    os.sync()
+                except AttributeError:
+                    # Python 2 doesn't expose this.  But set() is safe there.
+                    pass
+                try:
+                    origin.push(refspec="master:master")
+                except GitCommandError:
+                    raise BackendError(status_code=500,
+                                       reason="Internal Server Error",
+                                       content="Git push to " +
+                                       userdict["github_repo"] + " failed.")
+                retdict = {"repo_url": remote_url}
+                return retdict
+
+        def create_github_repository(auth, userdict):
+            """Create new repo at GH."""
+            ghub = github3.login(auth["username"], token=auth["password"])
+            try:
+                ghub.me()
+            except github3.exceptions.AuthenticationFailed:
+                raise BackendError(status_code=401,
+                                   reason="Bad credentials",
+                                   content="Github login failed.")
+            namespc = userdict["github_repo"]
+            org, name = namespc.split('/')
+            desc = ""
+            if "description" in userdict:
+                desc = userdict["description"]
+            gh_org = None
+            for gorg in ghub.organizations():
+                if gorg.login == org:
+                    gh_org = gorg
```

Comment:
How about `larry, moe, curly` here instead of `gh_org, gorg, org`, it would actually be easier to understand.